### PR TITLE
Have ActivityPrompt.RepromptDialogAsync call OnPromptAsync with isRetry set to false

### LINKED ISSUE #2449: Inconsistency about whether resuming a prompt should count as a retry
## Version
master branch

## Describe the bug

In [Prompt.cs](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.Dialogs/Prompts/Prompt.cs#L182), `ResumeDialogAsync` calls `RepromptDialogAsync` which calls `OnPromptAsync` with `isRetry` set to `false`:

```c#
public override async Task RepromptDialogAsync(ITurnContext turnContext, DialogInstance instance, CancellationToken cancellationToken = default(CancellationToken))
{
    var state = (IDictionary<string, object>)instance.State[PersistedState];
    var options = (PromptOptions)instance.State[PersistedOptions];
    await OnPromptAsync(turnContext, state, options, false, cancellationToken).ConfigureAwait(false);
}
```

However, in [ActivityPrompt.cs](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ActivityPrompt.cs#L166), `RepromptDialogAsync` calls `OnPromptAsync` with `isRetry` set to `true`:

```c#
public override async Task RepromptDialogAsync(ITurnContext turnContext, DialogInstance instance, CancellationToken cancellationToken = default(CancellationToken))
{
    var state = (IDictionary<string, object>)instance.State[PersistedState];
    var options = (PromptOptions)instance.State[PersistedOptions];
    await OnPromptAsync(turnContext, state, options, true, cancellationToken).ConfigureAwait(false);
}
```

I know it's a rare case when a prompt would be resumed, but in such a case should the retry prompt be used or not?

[bug]

### PR DESCRIPTION
**Pull request by v-kydela**
Fixes #2449

## Description
This fixes an inconsistency between Prompt.cs and ActivityPrompt.cs. In `ActivityPrompt`, resuming the prompt counted as a retry while in `Prompt` it did not. This is because of the `isRetry` parameter of `OnPromptAsync`.

Here is my reasoning to have them both be false rather than both be true. The kind of situation when a prompt would be resumed is when a user types "help" and then a help dialog is added to the stack and then ends. It makes more sense for the prompt to then say "Please enter your age" than to say "I'm sorry I couldn't understand that. Enter a number between 1 and 150."

In Python `is_retry` is already false in both cases so this choice helps with parity.

## Specific Changes

  - **Behavior change:** activity prompts now use [`PromptOptions.Prompt`](https://docs.microsoft.com/id-id/dotnet/api/microsoft.bot.builder.dialogs.promptoptions.prompt#Microsoft_Bot_Builder_Dialogs_PromptOptions_Prompt) instead of [`PromptOptions.RetryPrompt`](https://docs.microsoft.com/id-id/dotnet/api/microsoft.bot.builder.dialogs.promptoptions.retryprompt#Microsoft_Bot_Builder_Dialogs_PromptOptions_RetryPrompt) when [`RepromptDialogAsync`](https://docs.microsoft.com/id-id/dotnet/api/microsoft.bot.builder.dialogs.dialog.repromptdialogasync) is called.

See also: https://github.com/microsoft/botbuilder-js/pull/2431

### CODE REVIEW

**Comment by v-kydela** [comment]
Okay so the failing test was a little strange even before my change. Six replies got enqueued but only three were asserted. This worked because `TestFlow` doesn't throw any exceptions when there are leftover activities in the test adapter's queue. Anyway, I edited the test to make all six assertions, and also to account for the new `ResumeDialogAsync` behavior.

File: libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ActivityPrompt.cs [hunk 1]
```diff
@@ -189,7 +189,7 @@ public override async Task RepromptDialogAsync(ITurnContext turnContext, DialogI
         {
             var state = (IDictionary<string, object>)instance.State[PersistedState];
             var options = (PromptOptions)instance.State[PersistedOptions];
-            await OnPromptAsync(turnContext, state, options, true, cancellationToken).ConfigureAwait(false);
+            await OnPromptAsync(turnContext, state, options, false, cancellationToken).ConfigureAwait(false);
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ActivityPrompt.cs:192]: >false [](start = 61, length = 5)

This is a behavior change which could change the behavior of bots.  Please make sure this is documented in the change list in a way for customers to know that this happened.

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ActivityPromptTests.cs [hunk 2]
```diff
@@ -193,8 +193,11 @@ public async Task ActivityPromptResumeDialogShouldReturnDialogEndOfTurn()
             })
             .Send("hello")
             .AssertReply("please send an event.")
+            .AssertReply("please send an event.")
```
Comment (tomlm) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/ActivityPromptTests.cs:196]: we want the bot to be repeating itself?

Comment (v-kydela) [reply tests/Microsoft.Bot.Builder.Dialogs.Tests/ActivityPromptTests.cs:196]: In a real-world situation, `ResumeDialogAsync` would be called in response to a dialog on top of this dialog getting popped off of the stack. The extra dialog has been omitted here for simplicity, but you can imagine some dialog happening in between the two instances of "please send an event."

**Comment by cleemullins** [comment]
@Stevenic Can you take a quick look at this?

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ActivityPromptTests.cs (hunk 2 above)
Comment (v-kydela) [reply tests/Microsoft.Bot.Builder.Dialogs.Tests/ActivityPromptTests.cs:196]: @tomlm - I will assume you are satisfied with my answer

**Comment by EricDahlvang** [comment]
replaced by: https://github.com/microsoft/botbuilder-dotnet/pull/4298

**Pull request closed without merging**